src/main/java/teammates/ui/webapi/output/SessionResultsData.java

```diff
@@ -0,0 +1,259 @@
+import teammates.common.datatransfer.questions.FeedbackResponseDetails;
+import teammates.common.util.Const;
+
+/**
+ * API output format for session results, including statistics.
+ */
+public class SessionResultsData extends ApiOutput {
+
+    private static final String REGEX_ANONYMOUS_PARTICIPANT_HASH = "[0-9]{1,10}";
+
+    private final List<QuestionOutput> questions = new ArrayList<>();
+
+    public SessionResultsData(FeedbackSessionResultsBundle bundle, InstructorAttributes instructor) {
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionsWithResponses =
+                bundle.getQuestionResponseMapSortedByRecipient();
+
+        questionsWithResponses.forEach((question, responses) -> {
+            FeedbackQuestionDetails questionDetails = question.getQuestionDetails();
+            QuestionOutput qnOutput = new QuestionOutput(question.questionNumber, question.questionType,
+                    questionDetails.getQuestionText(),
+                    questionDetails.getQuestionResultStatisticsJson(responses, question, instructor.email, bundle, false));
+
+            List<ResponseOutput> allResponses = buildResponses(responses, bundle);
+            for (ResponseOutput respOutput : allResponses) {
+                qnOutput.allResponses.add(respOutput);
+            }
+
+            questions.add(qnOutput);
+        });
+    }
+
+    public SessionResultsData(FeedbackSessionResultsBundle bundle, StudentAttributes student) {
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionsWithResponses =
+                bundle.getQuestionResponseMapSortedByRecipient();
+
+        questionsWithResponses.forEach((question, responses) -> {
+            FeedbackQuestionDetails questionDetails = question.getQuestionDetails();
+            QuestionOutput qnOutput = new QuestionOutput(question.questionNumber, question.questionType,
+                    questionDetails.getQuestionText(),
+                    questionDetails.getQuestionResultStatisticsJson(responses, question, student.email, bundle, true));
+
+            if (questionDetails.isIndividualResponsesShownToStudents()) {
+                List<ResponseOutput> allResponses = buildResponses(question, responses, bundle, student);
+                for (ResponseOutput respOutput : allResponses) {
+                    if ("You".equals(respOutput.giver)) {
+                        qnOutput.responsesFromSelf.add(respOutput);
+                    } else if ("You".equals(respOutput.recipient)) {
+                        qnOutput.responsesToSelf.add(respOutput);
+                    } else {
+                        qnOutput.otherResponses.add(respOutput);
+                    }
+                }
+            }
+
+            questions.add(qnOutput);
+        });
+    }
+
+    public List<QuestionOutput> getQuestions() {
+        return questions;
+    }
+
+    private static String removeAnonymousHash(String identifier) {
```

Comment:
Why we need to remove anonymous hash here? I did not touch result layer for a long time :P

Reply:
Neither have I, actually.

https://github.com/TEAMMATES/teammates/blob/1ac836dae78ccbc4d1e76924d08511a7ddbdd1b6/src/main/java/teammates/ui/pagedata/StudentFeedbackResultsPageData.java#L165-L187

May not be necessary when the intent is instructor results, but it can only be known later.